RUFAS/biophysical/animal/ration/ration_optimizer.py

```diff
@@ -727,52 +755,69 @@ def attempt_optimization(
         """
         ration_config = RationConfig(requirements, pen_available_feeds, pen_average_body_weight)
 
-        if previous_ration:
-            x0: List[float] = []
-            prev_ration = previous_ration.copy()
-            for key, value in prev_ration.items():
-                if key not in ["status", "objective"]:
-                    x0.append(value)
-        else:
-            n = len(ration_config.price_list)
-            x0 = [1] + [random.random() * 10 for _ in range(n - 1)]
+        initial_decision_vector = np.array(self._build_initial_value(previous_ration, ration_config), dtype=float)
 
-        set_bounds = list(
-            zip([(lim) for lim in ration_config.feed_minimum_list], [(lim) for lim in ration_config.feed_maximum_list])
-        )
-        for i in range(0, len(set_bounds)):
-            if x0[i] < set_bounds[i][0] or x0[i] > set_bounds[i][1]:
-                x0[i] = np.clip(x0[i], set_bounds[i][0], set_bounds[i][1])
+        bounds = self._build_bounds(ration_config)
 
+        initial_decision_vector = self._check_initial_bounds(bounds, initial_decision_vector)
         arguments = (ration_config,)
+
         self.set_constraints(arguments=arguments)
 
-        if animal_combination is AnimalCombination.LAC_COW:
-            constraints_to_use = self.cow_constraints
-        elif animal_combination in [
-            AnimalCombination.GROWING,
-            AnimalCombination.CLOSE_UP,
-            AnimalCombination.GROWING_AND_CLOSE_UP,
-        ]:
-            constraints_to_use = self.heifer_constraints
-        else:
-            raise ValueError("Invalid animal combination: " + str(animal_combination))
+        constraints_to_use = self._select_constraints(animal_combination)
 
         optimized_ration_attempt = minimize(
             self.objective,
-            x0,
+            initial_decision_vector,
             method="SLSQP",
-            bounds=set_bounds,
+            bounds=bounds,
             constraints=constraints_to_use,
             args=arguments,
         )
```

Comment:
Thank you for making this more readable! This function is still confusing to me exactly what it's doing but it seems like the logic is the same but using human-readable language and just wanted to say I appreciate the work to make this more accessible.